api/src/main/java/org/apache/iceberg/ChangeScan.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.List;
+
+/**
+ * API for configuring a table scan for changes.
+ */
+public interface ChangeScan extends Scan<TableScan> {
+  /**
+   * Instructs this scan to look for changes starting from a particular snapshot (inclusive).
+   * <p>
+   * If the start snapshot is not configured, it is defaulted to the oldest ancestor
+   * of the end snapshot (inclusive).
+   *
+   * @param fromSnapshotId the start snapshot ID (inclusive)
+   * @return this for method chaining
+   */
+  ChangeScan fromSnapshotInclusive(long fromSnapshotId);
+
+  /**
+   * Instructs this scan to look for changes starting from a particular snapshot (exclusive).
+   * <p>
+   * If the start snapshot is not configured, it is defaulted to the oldest ancestor
+   * of the end snapshot (inclusive).
+   *
+   * @param fromSnapshotId the start snapshot ID (exclusive)
+   * @return this for method chaining
+   */
+  ChangeScan fromSnapshotExclusive(long fromSnapshotId);
+
+  /**
+   * Instructs this scan to look for changes up to a particular snapshot (inclusive).
+   * <p>
+   * If the end snapshot is not configured, it is defaulted to the current table snapshot (inclusive).
+   *
+   * @param toSnapshotId the end snapshot ID (inclusive)
+   * @return this for method chaining
+   */
+  ChangeScan toSnapshot(long toSnapshotId);
+
+  /**
+   * Instructs this scan to assign equality delete files to matching data files,
+   * which may be expensive if equality deletes are not scoped to particular partitions.
+   *
+   * @return this for method chaining
+   */
+  ChangeScan resolveEqualityDeletes();
+
+  /**
+   * Plans scan tasks for all file changes.
+   *
+   * @return this for method chaining
+   */
+  FileChangeIndex planFiles();
```

Comment:
The original `planFiles` did not work well as we have different types of files to plan and those will have to be read and processed differently. Any other ideas are welcome!

Reply:
this makes sense to me. original `planFiles` is not flexible enough to cover the more complex changelog scan

Reply:
One alternative would be to add new `FileScanTask`s for different types but that's going to be ugly to navigate and consume, in my view.

Reply:
I reconsidered this point as I am thinking of adapting the readers instead.

Reply:
Resolving as it no longer applies.